routes/user/index.js

```diff
@@ -0,0 +1,55 @@
+const express = require( 'express' )
+const router = express.Router()
+const passport = require( '../../configuration/passport' )
+const User = require( '../../database/user' )
+
+router.post( '/signup', ( request, response, next ) => {
+  const { email, password } = request.body
+
+  User.findByEmail( email )
+    .then( user => {
+      request.flash( 'message', 'Email already registered.' )
+      response.redirect( '/' )
+    })
+    .catch( error => {
+      const { hash, salt } = User.generateHash( password )
+
+      User.create( email, hash, salt )
+        .then( user => {
+          request.login( user, error => {
+            if( error ) {
+              return next( error )
+            } else {
+              response.redirect( '/users/profile' )
+            }
+          })
+        })
+        .catch( error => {
+          console.log( 'error creating user', error )
+        })
+    })
+})
+
+const authConfig = {
+  successRedirect: '/users/profile',
+  failureRedirect: '/',
+  failureFlash: true
+}
+
+router.post( '/login', passport.authenticate( 'local', authConfig ))
+
+router.get( '/logout', ( request, response ) => {
```

Comment:
Please change this to a `POST` request. Technically you're changing the state of the program, which a `GET` request should not do.